### DISCUSSION TITLE
Moved to Ausgrid Two Way Tarriff - but API not reflecting correct feed in ?

### DISCUSSION THREAD

**Discussion by craigcurtin-dev** [amberelectric/public-api#135] [General]
Hey Guys,

I was advised this morning by one of the customer support team that i have been moved onto the new Two Way Ausgrid Tarriffs.

I had to wait a few hours and the app (Android) then reflect that as did the Website after a couple of hours.

However my custom built Automations using Node Red are not reflecting the new Feed in price

I have generated a new Token and added that to my scripts and they are pulling data correctly - just not the correct data

Is there a different/updated API for the Two Way Tarriff or something else i need to do ?

Craig

**Comment by madpilot** [comment]
Hey Craig,

I actually just sent you a response to your help desk enquiry. But to repeat here for others: no need for a new API, the prices in app and on the API are the same.

You can double check the tariff code returned from the data in `/sites`. If it does say what you expect, let me know, and I can dig in further

**Reply by craigcurtin-dev** [reply]
Aah - just looking now - they have given me a new SiteID after i churned back to you (the old ID still works - go figure)

The new ID is showing i have two tarriffs - will this mean any difference in terms of data structure returned ?

Craig

**Reply by craigcurtin-dev** [reply]
Yep just dug in and there are now extra fields/objects for the 2nd tarriff - instead of the just the spot price  - if i am reading that correctly ?

I might just do a quick kludge now for my automations and put in the additional amount as a hard coded figure during the peak exports and then come back and rework my data extracts later

Craig

**Reply by madpilot** [reply]
The `pricePerKwh` field? Always been there - that is the price you want to use.

**Reply by craigcurtin-dev** [reply]
hmm - ok will play some more and report back

**Reply by madpilot** [reply]
The extra data in the `tariff` object just tells you if it's currently peak rate or not.

Hmmm. Not sure I've tested that with the two way feed in tariff. I'll check it's correct tomorrow.

**Reply by craigcurtin-dev** [reply]
OK Here is a screen shot

![image](https://user-images.githubusercontent.com/54323535/227105810-b1f20836-888d-40b8-a526-4d501ee3190b.png)

Not what i am expecting in the Spot Price ?

Craig

**Reply by craigcurtin-dev** [reply]
So basically my Javascript/NodeRed pulls down all the prices that are available and then stores them in arrays and builds a memory table of those for me to calculate etc - so it looks like the spotPerKwh is in the feedin channel which is what i expect - but i would expect the spot to be the non-signed of the perKwh price ?

Craig

**Reply by madpilot** [reply]
No - `perKwh` is the right price. `spotPerKwh` is the raw market value, not the price you get paid.

**Reply by madpilot** [reply]
The API is always from the perspective of imports to your house, so -40.57c means you get PAID 40.57c to export your power.

**Reply by craigcurtin-dev** [reply]
Aah - OK - will change my code to look at that - must have just been lucky all the way along !

thanks for the help

Craig

**Reply by madpilot** [reply]
>  so it looks like the spotPerKwh is in the feedin channel which is what i expect

Are you expecting the market price, or the price you get paid? `spotPerKwh` is the market price, `perKwh` is what you get paid.

> non-signed of the perKwh price

No, it's `-1 * perKwh`. A positive `perKwh` indicates you are paying for generation

**Reply by madpilot** [reply]
> must have just been lucky all the way along!

For 90% of feed-in tariffs, the two prices are basically the same. It's only these two-way tariffs and premium feed-in tariffs where the number is significantly different.

**Reply by craigcurtin-dev** [reply]
Aaah good point - better not do the absolute based on what you have said above - will have to cater for that now as well

Yeah more interested in what i get paid - so i can get my battery autmation happening correctly

Thanks for all the help

Craig
